proto/penumbra/penumbra/core/transaction/v1alpha1/transaction.proto

```diff
@@ -31,20 +31,32 @@ message EffectHash {
 message TransactionBody {
   // A list of actions (state changes) performed by this transaction.
   repeated Action actions = 1;
+  // Parameters determining if a transaction should be accepted by this chain.
+  TransactionParameters transaction_parameters = 2;
+  // The transaction fee.
+  crypto.v1alpha1.Fee fee = 3;
+  // Detection data for use with Fuzzy Message Detection
+  DetectionData detection_data = 4;
+  // An optional encrypted memo. It will only be populated if there are
+  // outputs in the actions of this transaction. 528 bytes.
+  optional bytes encrypted_memo = 5;
```

Comment:
Can we make this a sub-message (maybe `MemoData`?) rather than an `optional bytes`?  That way, we'll have better extensibility (if we ever want to add fields, we can add it to the `MemoData`) and it will be easier to do packetwise effect hashing (we can just hash the `MemoData`).